doc/tutorials/intermediate/structure_data_store.md

```diff
@@ -0,0 +1,330 @@
+# Structure with a DataStore
+
+In this tutorial you will learn how to use the `DataStore` *design pattern*:
+
+- The `DataStore` design pattern has proven quite successful across a wide range of use cases
+- The `DataStore` component receives `data` and `filters`. Based on the `data` and `filters` it *transforms* the data
+- One or more `View` components consumes the *transformed* data.
+- The components can be (re-)used for data exploration in notebooks and for building apps
+
+:::{admonition} Note
+When building a larger application we generally recommend using a *class based construction* and following best practices for object-oriented programming, specifically composition.
+:::
+
+```{pyodide}
+import panel as pn
+pn.extension('tabulator', 'vega', throttled=True)
+```
+
+<div class="header-box" style="box-shadow: rgba(50, 50, 93, 0.25) 0px 6px 12px -2px, rgba(0, 0, 0, 0.3) 0px 3px 7px -3px; padding: 5px 10px; border-left: 4px solid green;">
+
+## Install the dependencies
```

Comment:
My context is that we have users in our company that uses all kinds of frameworks. Like Streamlit, Dash, Panel, FastAPI+React. And in experience apps end up containing lots of functionality because users need it.

Thus my recommended structure as below

- `data`: data extracstions from databases in a `data` file or `module` depending on complexity. It's ok to import Param. But not Panel.
- `transformations`: pandas data transformation in a `transformations` data file or `module`. Its ok to import Param. But not Panel.
- `plots`: hvPlot/ HoloView, Matplotlib, Plotly plots from primarely Pandas Series and DataFrames in a `plots` file or module. Don't include Param or Panel here. Pure functions only.
- `components`: Break down your app UI into components. Implement the components using Param and Panel in a `components` file or module. Both the `DataStore` and the `View`s would fit in this category.
- `app`: Put things together in an `app` file.